pkg/adapter/kubernetes/componentcr/componentInstallationRepo_test.go

```diff
@@ -357,6 +361,29 @@ func Test_componentInstallationRepo_Create(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("should return error on convert component installation error", func(t *testing.T) {
+		// given
+		componentClientMock := newMockComponentRepo(t)
+		sut := componentInstallationRepo{
+			componentClient: componentClientMock,
+		}
+		componentInstallation := &ecosystem.ComponentInstallation{
+			Name:    testComponentName,
+			Version: testVersion1,
+			PackageConfig: map[string]interface{}{
+				"deployNamespace": map[string]string{"no": "string"},
+			},
+		}
+
+		// when
+		err := sut.Create(testCtx, componentInstallation)
+
+		// then
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "failed to convert component installation \"k8s/my-component\"")
+		assert.IsType(t, err, &domainservice.InternalError{})
```

Comment:
In other parts of the operator we just wrap `InternalError` with a generic error so that we can add more context. My thought was, that this could happen in the repository too but it is ok, if you leave it this way.

I checked sth. like this with this in the past:
```
expectedErr := &domainservice.NotFoundError{}
assert.ErrorAs(t, err, &expectedErr)
```
For `NotFoundErrors` we also just introduced a util function `domainservice.IsNotFoundError()` for this as we checked this so often.